Duplicati/Library/Main/Operation/DeleteHandler.cs

```diff
@@ -218,37 +225,30 @@ private DateTime[] GetFilesetsToDelete(Database.LocalDeleteDatabase db, DateTime
             // and remove oldest backups while there are still more backups than should be kept as specified via option
             var backupsRemaining = sortedAllBackups.Except(toDelete).ToList();
             var fullVersionsToKeep = m_options.KeepVersions;
-            var fullVersionsKeptCount = 0;
             if (fullVersionsToKeep > 0 && fullVersionsToKeep < backupsRemaining.Count)
             {
-                // keep the number of full backups specified in fullVersionsToKeep.
-                // once the last full backup t okeep is found, also keep the partials immediately after it the full backup.
-                // add the remainder of full and partial backups to toDelete
-                bool foundLastFullBackupToKeep = false;
-                foreach (var backup in backupsRemaining)
+                int fullVersionsKept = 0;
+                ISet<DateTime> intermediatePartials = new HashSet<DateTime>();
+
+                // Enumerate the collection starting from the most recent full backup.
+                foreach (DateTime backup in backupsRemaining.SkipWhile(x => !db.IsFilesetFullBackup(x)))

```

Comment:
With this we do not get partial backups, but within the foreach you try to remove partial backups?

Reply:
Sorry, I'm not sure I understand.  `SkipWhile`will skip elements until the first occurrence of `false`, and then return the rest of the sequence.  So, I think this will only skip partial backups at the beginning of the sequence?

Reply:
You're right, I misread the description of skipwhile.